### ISSUE TITLE
DOOR record edits can show up as "dirty"

### ISSUE THREAD

**Issue by BlackPete84** [TES5Edit/TES5Edit#1317]
**What is the version of xEdit you are using?**
SSEEdit 4.1.5 Experimental

**Describe the bug**
DOOR record edits can seemingly show up as dirty edits under a specific circumstance. Since I don't understand enough about how xEdit is programmed, the best description I can give you is under "Steps to Reproduce" below. Please also refer to the linked screenshot at the end of the report.

**To reproduce**
1. Start the Skyrim Special Edition Creation Kit. Make sure that both Skyrim.esm and Update.esm are selected. You can also select the three DLC and the four free plugins that now come with SSE, but it likely isn't necessary for this purpose.
2. When it is done loading, click on the Edit dropdown menu.
3. Choose "Find Text".
4. When the box pops up choose the "Objects, Etc" tab.
5. Enter 000DE023 in the Search Text field, and double click on the result that appears below.
6. The preview window should show a trap door. Double click on the trap door in the preview window, which will bring up a Reference window.
7. Select the "Ownership" tab in the Reference window.
8. In the "Faction" dropdown menu change the entry from "HagravenFaction" to "NONE". You'll need to scroll up to find the "NONE".
9. The dropdown menu to the left labeled "NPC" will now be useable. Change the entry from "NONE" to "dunPOIWitchAnise".
10. Click "OK" in the Reference box that you've been using since Step 6.
11. If the "Text Search" box (from Step 4) is still open, close it. Save the plugin. The file name obviously shouldn't matter as long as it doesn't overwrite any of your existing ones. The Creation Kit can now be closed.
12. When you open up the plugin you just created in the Creation Kit using SSEEdit, you should notice that two references are edited: 000DE023 and 000DDFB0.
13. 000DE023 will show the expected edit. Since 000DE023 is linked to a different trap door (000DDFB0) in another cell, it will also show up as edited. However, 000DDFB0 shows up as a dirty edit for me.  Please see the screenshot link below.

**Expected behavior**
Record would properly show up as edited and not as a dirty edit. At least that's what I would expect. The person who is reading this report will likely know for certain.

**Screenshots**
https://postimg.cc/NKMfXJjW

**Additional context**
The information provided above should be enough, but you can let me know if it isn't.

**Comment by jbostrus** [comment]
From your expected behavior note I don't really understand what you mean. You expect the record to show up as edited and not a dirty edit. Exactly what is saying something is a dirty edit? All color coding is indications that the record is edited. An override of any kind is an edit. An ITM is an edit. xEdit does not make any distinction or suggestion on its own that something is a "dirty" edit.

xEdit just shows the data as it is in the file. The context of something being a dirty edit is how the data is interpreted. xEdit does not have any suggestion from it that identifies something as being considered a dirty edit. Generally a dirty edit is referred to as some kind of change made to the record that was unintentional. Sometimes it is benign, such as moving an object 1 pixel. 

In regards to your specific example case. If you look at 000DDFB0 you will see that Creation Kit decided to change the Location Reference XLRL subrecord. Since the original record did not have a location set, but CK now realizes that the object is within a location's bounds it decided to make an override and updates that. This is just how CK works. It has nothing to do with xEdit. You will also see the same change was implemented on the 000DE023 record you mentioned directly editing. Even though you did not explicitly make that change yourself.

In short, xEdit is doing what it's supposed to do. Showing you the data that is stored in the plugin. And identifying if, and how, it differs from any master data.

**Comment by BlackPete84** [comment]
Thanks for the reply and sorry about the confusion. When I used the term dirty edit, I actually meant that 000DDFB0 was being treated as Identical To Master (ITM) record in xEdit. I can see the change to the XLRL record, however, when using the "hide no conflict and empty rows" feature the edit is no longer visible [[screenshot](https://i.postimg.cc/T3520DKZ/xEdit.png)], which probably shouldn't be the case. Also, cleaning the plugin (using the quick auto-clean feature) results in xEdit removing the edit to 000DDFB0 altogether. Obviously this could be resolved by not cleaning the plugin, but under normal circumstances xEdit will retain legitimate edits after cleaning the plugin.

Hopefully the explanation above clarifies what I meant. This is a very specific case and probably would only affect a select few people, so I'll understand if you're not able to reproduce it or don't consider it to be something that needs to be addressed. Thanks for your time either way.

**Comment by jbostrus** [comment]
It is quite easy to reproduce. That isn't an issue. It's a matter of whether anyone wants to address it as an issue or not. 

That is not an ITM. An ITM means every single field entry is identical in every way. That is however a benign conflict. You chose to to hide **NO CONFLICT** and empty rows. The XLRL is designated as a benign conflict **IF** it is only added and the master does not have it present. Thus it is hidden by that option. The record is removed during cleaning because when it checks the conflict state, because it is a benign conflict it considers it "close enough" to an ITM and removes it because everything else within the record comes back as identical.

It's kind of a tricky situation there. While it technically could be an intentional edit. The bigger question at that point would be why? Particularly if that was the only change. As it would have no functional purpose. So on this since that particular element is benign, as long as nothing else is changed xEdit removes it.

**Comment by BlackPete84** [comment]
Okay, I get where you're coming from. I don't always get the technical terminology correct, as was the case here. If it is indeed the case that nobody wants to address it as issue, please go ahead and close this report. Thanks for your time and explanation; it is greatly appreciated.

**Comment by jbostrus** [comment]
Easier if you just close it yourself as there's only a handful of people who can do it otherwise. And they're generally pretty busy.
